binding/jms/src/test/java/io/tracee/binding/jms/TraceeMessageListenerTest.java

```diff
@@ -47,22 +47,31 @@ public Object answer(InvocationOnMock invocation) throws Throwable {
 
 	@Test
 	public void testDecodesFromMessage() throws Exception {
-		encodedContext.put("contextFromMessage","yes");
+		encodedContext.put(INVOCATION_ID_KEY, "yes");
 		unit.intercept(invocationContext);
-		assertThat(backend.getValuesBeforeLastClear(), hasEntry("contextFromMessage", "yes"));
+		assertThat(backend.getValuesBeforeLastClear(), hasEntry(INVOCATION_ID_KEY, "yes"));
+		verify(mdbLike).onMessage(message);
 	}
 
 	@Test
 	public void testCleansUpContextAfterProcessing() throws Exception {
-		encodedContext.put("contextFromMessage","yes");
+		encodedContext.put("contextFromMessage", "yes");
 		unit.intercept(invocationContext);
 		assertThat(backend.isEmpty(), is(true));
+		verify(mdbLike).onMessage(message);
+	}
+
+	@Test
+	public void testInvocationIdGeneratedIfAbsent() throws Exception {
+		unit.intercept(invocationContext);
```

Comment:
Got it. I'll push a patch later. And if that's OK to you. Please leave a message here, I'll rebase the commits for clearer commit history.